src/middlewared/middlewared/plugins/pool.py

```diff
@@ -696,9 +718,12 @@ def add(attr):
 
         dataset = await self.middleware.call('pool.dataset.query', [('id', '=', id)])
         if not dataset:
-            verrors.add('id', f'{id} does not exist', errno.ENOENT)
+            verrors.add('pool_dataset_update.id', f'{id} does not exist', errno.ENOENT)
```

Comment:
You should revert this back I think. `id` is a main schema, not within `pool_dataset_update`